File: rust/598.range-addition-ii.rs
/*
 * @lc app=leetcode id=598 lang=rust
 *
 * [598] Range Addition II
 */

// @lc code=start
impl Solution {
    pub fn max_count(m: i32, n: i32, ops: Vec<Vec<i32>>) -> i32 {
        let (mut m, mut n) = (m, n);
        for op in ops {
            m = m.min(op[0]);
            n = n.min(op[1]);
        }
        m * n
    }
}
// @lc code=end

File: javascript/598.range-addition-ii.js
/*
 * @lc app=leetcode id=598 lang=javascript
 *
 * [598] Range Addition II
 */

// @lc code=start
/**
 * @param {number} m
 * @param {number} n
 * @param {number[][]} ops
 * @return {number}
 */
var maxCount = function (m, n, ops) {
  for (const [_m, _n] of ops) {
    m = Math.min(m, _m);
    n = Math.min(n, _n);
  }
  return m * n;
};
// @lc code=end
